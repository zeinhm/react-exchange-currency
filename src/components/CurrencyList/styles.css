.currency-list {
  display: flex;
  justify-content: center;
  padding-top: 3rem;
}

.currency-list > section {
  border: 1px solid #999;
  display: flex;
  flex-direction: column;
  min-width: 18rem;
  min-height: 30rem;
}

.currency-list > section > header {
  padding: 1rem 1rem 0;
  border-bottom: 1px solid #999;
}

.currency-list > section > header > p {
  font-size: .865rem;
  font-style: italic;
}

.currency-list > section > header > div > p,
.currency-list > section > header > div > input {
  font-size: 1rem;
  font-weight: bold;
}

.currency-list > section > header > div {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 1rem 0;
}

.currency-list > section > header > div > input {
  width: 4rem;
}

.currency-list > section > header > div > input::-webkit-inner-spin-button,
.currency-list > section > header > div > input::-webkit-outer-spin-button{
  -webkit-appearance: none;
  margin: 0
}