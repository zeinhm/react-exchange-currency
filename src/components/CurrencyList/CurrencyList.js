import React from 'react'
import AddCurrency from '../AddCurrency/AddCurrency'
import CurrencyItems from '../CurrencyItems'
import './styles.css'

export default function CurrencyList(props) {

  const { 
    addCurrency,
    amount,
    currencies,
    deleteCurrency,
    onChange,
    selectedCurrency,
    setSelectedCurrency,
  } = props
  
  return (
    <main className='currency-list'>
      <section>
        <header>
          <p>USD - United States Dollars</p>
          <div>
            <p>USD</p>
            <input value={amount} onChange={e => onChange(e.target.value)} type="number"/>
          </div>
        </header>
        <CurrencyItems
          amount={amount}
          currencies={currencies}
          deleteCurrency={deleteCurrency}
        />
        <AddCurrency
          onSubmit={addCurrency}
          selectedCurrency={selectedCurrency}
          setSelectedCurrency={setSelectedCurrency}
        />
      </section>
    </main>
  )
}
