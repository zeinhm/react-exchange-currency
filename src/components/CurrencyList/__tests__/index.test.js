import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CurrencyList from '../CurrencyList';

jest.mock('../../../contexts', () => ({
  AppContext: {
    amount: 0,
    currencies: [],
    currencyRates: {},
    setAmount: jest.fn(),
    setCurrencies: jest.fn(),
    setCurrencyRates: jest.fn()
  },
}));

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: v => v,
  useEffect: cb => cb(),
  useState: v => [v === 'tes' ? v : !v, jest.fn()],
}));

describe('src/components/CurrencyList', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const props = {
      amount: 100,
      code: 'test',
      currencies: [{ code: 'test', rates: 100}],
      deleteCurrency: () => {},
      onChange: () => {}
    }
    const tree = shallow.render(<CurrencyList {...props} />);
    tree.props.children.props.children[0].props.children[1].props.children[1]
      .props.onChange({ target: { value: 'test' } })
    expect(tree).toMatchSnapshot();
  });
});
