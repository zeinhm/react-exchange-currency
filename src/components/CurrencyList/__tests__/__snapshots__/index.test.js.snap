// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/CurrencyList render 1`] = `
<main
  className="currency-list"
>
  <section>
    <header>
      <p>
        USD - United States Dollars
      </p>
      <div>
        <p>
          USD
        </p>
        <input
          onChange={[Function]}
          type="number"
          value={100}
        />
      </div>
    </header>
    <CurrencyItems
      amount={100}
      currencies={
        Array [
          Object {
            "code": "test",
            "rates": 100,
          },
        ]
      }
      deleteCurrency={[Function]}
    />
    <AddCurrency />
  </section>
</main>
`;
